src/components/common/modal.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useState } from 'react';
+import { Modal as NextUIModal } from '@nextui-org/react';
+
+const Modal = ({
+  children,
+  visible,
+  onClose,
+}: {
+  children: React.ReactNode;
+  visible: boolean;
+  onClose: () => void;
+}) => {
+  return (
+    <div>
+      <NextUIModal
+        scroll
+        width="600px"
```

Comment:
We might want to consider something more generic.

Reply:
Right, I initially added a css + HTML modal, but since I see we already are using NextUI this is more solid perhaps that what I put there. But I can all one later we more generic design.